docker/generic/start_proxy.py

```diff
@@ -518,6 +523,8 @@ def enforce_conflict_args(args):
     if len(port_flags) > 1:
         return "Multiple port flags {} are not allowed, please just use --listener_port flag".format(",".join(port_flags))
 
+    if args.ssl_port and args.ssl_port == args.listener_port:
```

Comment:
supporting both "ssl_port" and "listener_port" is confusing.

My intention was to make "listener_port" also supporting ssl.   Using a separate ssl flag to indicate it.

such as --ssl_server_cert_path to specify the path.  we don't hard code the path.

We may have to add other ssl flags too, such as ssl_version to indicate which versions to support

"force http redirect" is another feature. It is not very useful feature.  Normally, a container only open one port,  either http or https.   It is rare to open two ports; http and https

Reply:
I think it is better to have a design doc for this feature.

Reply:
Other features like "ssl_server_cert_path", "ssl_version" will be added later.

Eventually we need a flag, to indicate that this is a HTTPS server, and I think use ssl_port is totally fine, which is also compatible with ESPv1. 

I can delete the redirect part, if you think it is not useful. But I do see some examples of this.

WDYT?